### PR TITLE
[Merged by Bors] - feat: port Algebra.Star.Subalgebra

### PR DESCRIPTION
**Pull request by j-loreaux**
Note: Lean was continually crashing on me as I was porting this file, I believe sure to autoparams.

- [x] depends on: #2817

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#2817
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#2817~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

File: Mathlib/Algebra/Star/Subalgebra.lean [hunk 1]
```diff
@@ -0,0 +1,793 @@
...
+#align star_subalgebra.to_subalgebra_inj StarSubalgebra.toSubalgebra_inj
+
+theorem toSubalgebra_le_iff {S₁ S₂ : StarSubalgebra R A} :
+    S₁.toSubalgebra ≤ S₂.toSubalgebra ↔ S₁ ≤ S₂ :=
+  Iff.rfl
+#align star_subalgebra.to_subalgebra_le_iff StarSubalgebra.toSubalgebra_le_iff
+
+/-- Copy of a star subalgebra with a new `carrier` equal to the old one. Useful to fix definitional
+equalities. -/
+protected def copy (S : StarSubalgebra R A) (s : Set A) (hs : s = ↑S) : StarSubalgebra R A where
```
Comment (agjftucker) [inline Mathlib/Algebra/Star/Subalgebra.lean:120]:
```suggestion
protected def copy (S : StarSubalgebra R A) (s : Set A) (hs : s = ↑S) : StarSubalgebra R A :=
  { S.toSubalgebra.copy s hs with
    star_mem' := fun {a} => by simpa [hs] using S.star_mem' }
```
Can confirm the crashing with mathport's definition!

**Review by dupuisf — COMMENTED** [review]
Thanks!

bors d+

**Comment by j-loreaux** [comment]
bors r+

I'm fairly certain the crashing was due to autoparams issues as has been mentioned on Zulip recently.

**Pull request closed without merging**